src/contrib/serializers/Akka.Serialization.Hyperion.Tests/HyperionSerializerSetupSpec.cs

```diff
@@ -64,5 +71,42 @@ public void Setup_package_override_should_work()
             var adapter = appliedSettings.PackageNameOverrides.First();
             adapter("My.Hyperion.Override").Should().Be("My.Hyperion");
         }
+        
+        public class Foo
+        {
+            public Foo(string bar)
+            {
+                Bar = bar;
+            }
+
+            public string Bar { get; }
+        }
+        
+        public class FooSurrogate
+        {
+            public FooSurrogate(string bar)
+            {
+                Bar = bar;
+            }
+
+            public string Bar { get; }
+        }
+        
+        [Fact]
+        public void Setup_surrogate_should_work()
+        {
+            var setup = HyperionSerializerSetup.Empty
+                .WithSurrogates(new [] { Surrogate.Create<Foo, FooSurrogate>(
+                    foo => new FooSurrogate(foo.Bar + "."), 
+                    surrogate => new Foo(surrogate.Bar))
+                });
+            var settings = setup.ApplySettings(HyperionSerializerSettings.Default);
+            var serializer = new HyperionSerializer((ExtendedActorSystem)Sys, settings);
+
+            var expected = new Foo("bar");
+            var serialized = serializer.ToBinary(expected);
+            var deserialized = serializer.FromBinary<Foo>(serialized);
+            deserialized.Bar.Should().Be("bar.");
```

Comment:
Same feedback as before - need to guarantee that the `ISurrogate` actually gets used

Reply:
Done.